openvdb/cmd/openvdb_print.cc

```diff
@@ -11,13 +11,116 @@
 #include <openvdb/openvdb.h>
 #include <openvdb/util/logging.h>
 
+#include <openvdb/points/PointDataGrid.h>
+
+using namespace openvdb;
+using namespace openvdb::points;
 
 namespace {
 
 using StringVec = std::vector<std::string>;
 
 const char* INDENT = "   ";
 const char* gProgName = "";
+static const std::string sINDENT(INDENT);
+
+#if OPENVDB_ABI_VERSION_NUMBER >= 6
```

Comment:
I don't think you need to ABI switch this. I think the only methods you're using which are ABI6 only are `codecType` and `valueType`, both of which you can query from the descriptor using `valueType()` and `type()`

Reply:
Perhaps I'm doing this wrong but using codecType() and valueType() gives me different output:
`
myAttributeArray.codecType() : fxpt16
myDescriptor->valueType() : vec3s
`

Reply:
Ah you're looking for `type()`. The value type and codec are stored in a pair of strings which is typedef'd to a `NamePair` in VDB - this is the return value of `type()`. i.e:

```
 desc->type().first == desc->valueType();
 desc->type().second == array->codecType();
```

Reply:
Thanks, I will update that.